recipes-support/phytec-board-info/files/phytec-board-info.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 PHYTEC Messtechnik GmbH
+
+usage="
+Usage: $(basename $0) OPTION
+
+Print information about the current hardware.
+
+One of the following options can be selected at a time:
+  -c, --compatible        Device tree compatible of the machine
+  -m, --machine           Hardware machine name
+  -n, --simno             SIM number
+  -s, --serial            Serial number
+  -v, --machine-version   Hardware machine version
+"
+
+case "$1" in
+	-c|--compatible)
+		cut -f 1 -d '' /proc/device-tree/compatible
+		;;
+	-m|--machine)
+		cat /etc/hostname
+		;;
+	-v|--machine-version)
+		# Hardware version is hard to find out. The only way I can think of is
+		# using the machine name's version
+		rev /etc/hostname | cut -f 1 -d '-' | rev
+		;;
+	-s|--serial)
+		# Serial number is tied to MAC address of the hardware module
+		cut -f 1- -d ':' --output-delimiter '' /sys/class/net/eth0/address
```

Comment:
Somehow I am having difficulties when executing this option with the awsclient.
The awsclient only gets the first 2 characters when reading from stdout.
I suspect it has something to do with replacing the : with a null terminated empty string, but I am not sure what exactly is happening.
The following does work for me though.
`cat /sys/class/net/eth0/address | tr -d ':'`

Reply:
Ok. Will replace this with `tr`, which actually looks nicer. :+1: